src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java

```diff
@@ -217,4 +216,17 @@ String computeCrc32c(String input) {
         return outputs;
     }
 
+    private <V> boolean shouldRetry(Throwable throwable, V input) {
+        if (throwable == null) {
+            return false; // Did not find a retry-able exception
+        }
+        if (throwable instanceof DeadlineExceededException ||
+            throwable instanceof UnavailableException) {
+            logger.warn("Retry-able error in firestore future get - input: " +
+                input + " message: " + throwable.getMessage());
+            return true;
+        }
+        return shouldRetry(throwable.getCause(), input);
+    }
```

Comment:
What's the over under that we're gonna find a third one?

Reply:
I don't want to be the one to find out :)